### PR TITLE
feat(s3-deployment): control object access

### PR DESCRIPTION
**Pull request by sormy**
Add support for BucketDeployment accessControl property to aws-s3-deployment package.

This is needed to run, for example, `aws s3 sync` with `--acl bucket-owner-full-control`.
Without this feature there is no easy way (without hacking cdk nodes) to sync assets to bucket located in another account.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by sormy** [comment]
Is there anything else I can do for this CR to proceed?

**Review by kapilsd — DISMISSED** [review]

File: packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts [hunk 1]
```diff
@@ -325,6 +325,7 @@ test('system metadata is correctly transformed', () => {
     websiteRedirectLocation: 'example',
     cacheControl: [s3deploy.CacheControl.setPublic(), s3deploy.CacheControl.maxAge(cdk.Duration.hours(1))],
     expires: expiration,
+    accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts:328]: can you add a separate test case for each type of `BucketAccessControl`, so we can ensure that all types are correctly transformed by `toKebabCase()`?

File: packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts [hunk 2]
```diff
@@ -165,6 +166,14 @@ export interface BucketDeploymentProps {
    */
   readonly serverSideEncryptionCustomerAlgorithm?: string;
 
+  /**
+   * Sets the ACL for the object when the command is performed.
+   * If you use this parameter you must have the "s3:PutObjectAcl" permission included in the list of actions for your IAM policy.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts:171]: Which IAM policy should have this permission configured?

I'm not very familiar with this module but usually the CDK ensures that the permissions are correctly configured.

File: packages/@aws-cdk/aws-s3-deployment/README.md [hunk 3]
```diff
@@ -177,6 +178,7 @@ new s3deploy.BucketDeployment(this, 'DeployWebsite', {
   storageClass: StorageClass.INTELLIGENT_TIERING,
   serverSideEncryption: ServerSideEncryption.AES_256,
   cacheControl: [CacheControl.setPublic(), CacheControl.maxAge(cdk.Duration.hours(1))],
+  accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-s3-deployment/README.md:181]: We usually like to add a couple of sentences on what this feature, when it should be used and any caveats.

Some require a separate section but most just a couple of sentences.

File: packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts [hunk 4]
```diff
@@ -165,6 +166,14 @@ export interface BucketDeploymentProps {
    */
   readonly serverSideEncryptionCustomerAlgorithm?: string;
 
+  /**
+   * Sets the ACL for the object when the command is performed.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts:170]: Should this be "objects in this bucket"?

**Review by nija-at — CHANGES_REQUESTED** [review]
Please see my comments below.

@otaviomacedo - can you also take a look?

File: packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts (hunk 4 above)
Comment (sormy) [reply packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts]: I have updated docblock to match other properties related to system metadata.

File: packages/@aws-cdk/aws-s3-deployment/README.md (hunk 3 above)
Comment (sormy) [reply packages/@aws-cdk/aws-s3-deployment/README.md:184]: So `accessControl` is one of system metadata properties like `serverSideEncryption` and etc. It seems like there is no additional information provided for other system metadata properties here. I have added a note about how system metadata maps to `aws s3 sync` arguments and where additional information can be obtained.

File: packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts (hunk 2 above)
Comment (sormy) [reply packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts]: You are right. This permission is granted by construct. I have removed this sentence.

File: packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts (hunk 1 above)
Comment (sormy) [reply packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts:328]: I have added the test.

**Review by nija-at — APPROVED** [review]

**Pull request merged**